src/app/query/bitcoin/transaction/use-check-utxos.ts

```diff
@@ -36,6 +37,44 @@ export function filterOutIntentionalUtxoSpend({
   });
 }
 
+interface CheckInscribedUtxosByBestinslotArgs {
+  inputs: btc.TransactionInput[];
+  txids: string[];
+  client: ReturnType<typeof useBitcoinClient>;
+}
+
+async function checkInscribedUtxosByBestinslot({
+  inputs,
+  txids,
+  client,
+}: CheckInscribedUtxosByBestinslotArgs): Promise<boolean> {
+  /**
+   * @description Get the list of inscriptions moving on a transaction
+   * @see https://docs.bestinslot.xyz/reference/api-reference/ordinals-and-brc-20-and-bitmap-v3-api-mainnet+testnet/inscriptions
+   */
+  const inscriptionIdsList = await Promise.all(
+    txids.map(id => client.bestinslotInscriptionsApi.getInscriptionsByTransactionId(id))
+  );
+
+  const inscriptionIds = inscriptionIdsList.flatMap(inscription =>
+    inscription.data.map(data => data.inscription_id)
+  );
+
+  // fetch inscriptions by id to get relevant data
+  const responses = await Promise.all(
+    inscriptionIds.map(id => client.bestinslotInscriptionsApi.getInscriptionById(id))
+  );
+
+  // check if any of the utxos are inscribed
+  return responses.some(resp => {
+    return inputs.some(input => {
+      if (!input.txid) throw new Error('Transaction ID is missing in the input');
+      const idWithIndex = `${bytesToHex(input.txid)}:${input.index}`;
+      return resp.data.satpoint.includes(idWithIndex);
+    });
+  });
```

Comment:
Rather than a comments in this file, these could equally be functions with a descriptive name. 

A function is way better, because comments are much more likely to become stale & out of date when another developer changes the code.

Reply:
refactored here